libsolidity/interface/FileReader.h

```diff
@@ -0,0 +1,87 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <boost/filesystem.hpp>
+#include <map>
+#include <vector>
+
+#include <libsolidity/interface/ReadFile.h>
+
+namespace solidity {
+
+class FileReader
+{
+public:
+	using StringMap = std::map<std::string, std::string>;
+
+	FileReader(
+		boost::filesystem::path _basePath,
+		std::vector<boost::filesystem::path> _allowedDirectories
+	):
+		m_basePath(std::move(_basePath)),
+		m_allowedDirectories(std::move(_allowedDirectories)),
+		m_sourceCodes(),
+		m_fullPathMapping()
+	{}
+
+	FileReader(): FileReader({}, {}) {}
+
+	boost::filesystem::path const& basePath() const noexcept { return m_basePath; }
```

Comment:
Probably because the codebase uses exceptions heavily :)

Reply:
Also `noexcept` is fairly in this way. And since we're returning just a cref it is safe to explictly state that. It is common practice for me to make noexcept what is meant to be noexcept. But I'm fine in not doing that? :)

Reply:
I'm fine with that, just wondering if I should start doing that too. So far I just had the impression that we don't care about it too much but if we do, I'll start using it.